packages/db-base/lib/tools.js

```diff
@@ -15,32 +16,40 @@ function getControllerDir() {
             /* not found */
         }
     }
+
     // Apparently, checking vs null/undefined may miss the odd case of controllerPath being ""
     // Thus we check for falsyness, which includes failing on an empty path
     if (!controllerPath) {
-        const checkPath = path.normalize(path.join(__dirname, '../..'));
-        const pathParts = checkPath.split(path.sep);
-        while (pathParts.length) {
-            const tryPath = pathParts.join(path.sep);
-            if (fs.existsSync(path.join(tryPath, 'lib/tools.js'))) {
-                controllerPath = tryPath;
-                break;
+        let checkPath = path.join(__dirname, '../..');
+        // Also check in the current check dir (along with iobroker.js-controller subdirs)
+        possibilities.unshift('');
+        outer: while (true) {
+            for (const pkg of possibilities) {
+                try {
+                    const possiblePath = path.join(checkPath, pkg);
+                    if (fs.existsSync(path.join(possiblePath, 'lib/tools.js'))) {
+                        controllerPath = possiblePath;
+                        break outer;
+                    }
+                } catch (_a) {
+                    // not found, continue with next possiblity
+                }
             }
-            // Mainly for local development cases
-            if (fs.existsSync(path.join(tryPath, 'iobroker.js-controller/lib/tools.js'))) {
-                controllerPath = path.join(tryPath, 'iobroker.js-controller');
-                break;
-            }
-            if (fs.existsSync(path.join(tryPath, 'ioBroker.js-controller/lib/tools.js'))) {
-                controllerPath = path.join(tryPath, 'ioBroker.js-controller');
+
+            // Controller not found here - go to the parent dir
+            const newPath = path.dirname(checkPath);
+            if (newPath === checkPath) {
+                // We already reached the root dir, abort
                 break;
             }
-            pathParts.pop();
+            checkPath = newPath;
         }
+        // ??? What is this supposed to do ???
         if (controllerPath && !fs.existsSync(controllerPath)) {
             controllerPath = null;
         }
     } else {
+        // ??? What is this supposed to do ???
         controllerPath = path.dirname(controllerPath);
```

Comment:
And this makes even less sense IMO

Reply:
It makes perfect sense. YOu come there when the require lookip was working, then controllerpath contains the "index.js" in the end ... so you need to make a dir out of it

Reply:
Ok, I didn't think about the `index.js` part. Should we rather try to resolve the `package.json`, so we know the resolved file is definitely in the project root (just in case we ever change that and forget this check)? Example for "fs-extra":
![grafik](https://user-images.githubusercontent.com/17641229/93929719-8a559e80-fd1c-11ea-84ff-d910e11f7da0.png)

Reply:
like the idea